packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -152,194 +232,274 @@ class MouseTracker extends ChangeNotifier {
-    if (event.kind != PointerDeviceKind.mouse) {
+    if (event.kind != PointerDeviceKind.mouse)
+      return;
+    if (event is PointerSignalEvent)
       return;
-    }
     final int device = event.device;
-    if (event is PointerAddedEvent) {
-      _addMouseDevice(device, event);
-    } else if (event is PointerRemovedEvent) {
-      _removeMouseDevice(device, event);
-    } else if (event is PointerHoverEvent) {
-      final _MouseState mouseState = _guaranteeMouseState(device, event);
-      final PointerEvent previousEvent = mouseState.mostRecentEvent;
-      mouseState.mostRecentEvent = event;
-      if (previousEvent is PointerAddedEvent || previousEvent.position != event.position) {
-        // Only send notifications if we have our first event, or if the
-        // location of the mouse has changed
-        _checkDeviceUpdates(device: device);
-      }
-    }
-  }
+    final _MouseState existingState = _mouseStates[device];
+    if (!_shouldMarkStateDirty(existingState, event))
+      return;
 
-  bool _scheduledPostFramePositionCheck = false;
-  // Schedules a position check at the end of this frame.
-  // It is only called during a frame during which annotations have been added.
-  void _scheduleMousePositionCheck() {
-    // If we're not tracking anything, then there is no point in registering a
-    // frame callback or scheduling a frame. By definition there are no active
-    // annotations that need exiting, either.
-    if (!_scheduledPostFramePositionCheck) {
-      _scheduledPostFramePositionCheck = true;
-      SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
-        _checkAllDevicesUpdates();
-        _scheduledPostFramePositionCheck = false;
-      });
-    }
+    final PointerEvent previousEvent = existingState?.latestEvent;
+    _updateDevices(
+      targetEvent: event,
+      handleUpdatedDevice: (_MouseState mouseState, LinkedHashSet<MouseTrackerAnnotation> previousAnnotations) {
+        assert(mouseState.device == event.device);
+        _dispatchDeviceCallbacks(
+          lastAnnotations: previousAnnotations,
+          nextAnnotations: mouseState.annotations,
+          handledEvent: previousEvent,
+          unhandledEvent: event,
+          trackedAnnotations: _trackedAnnotations,
+        );
+      },
+    );
   }
 
-  // Collect the latest states of the given mouse device `device`, and call
-  // interested callbacks.
+  // Find the annotations that is hovered by the device of the `state`.
   //
-  // The enter or exit events are called for annotations that the pointer
-  // enters or leaves, while hover events are always called for each
-  // annotations that the pointer stays in, even if the pointer has not moved
-  // since the last call. Therefore it's caller's responsibility to check if
-  // the pointer has moved.
+  // If the device is not connected or there are no annotations attached, empty
+  // is returned without calling `annotationFinder`.
```

Comment:
No need for backticks here.

Reply:
Is it ok to use backticks in docs so that it's clearer that I'm referring to a symbol?

Reply:
Actually I've found quite a few existing examples of [using backticks](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/heroes.dart#L329) or [using square brackets](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/reorderable_list.dart#L183) in normal comments.

Reply:
Well, just because they're common, doesn't make them right. :-) Since they're not checked for correctness by dartdoc, and sometimes are not searched during IDE refactors, references can become invalid over time, so not having the markup and using `//` serves as a reminder that if that API or comment ever goes into the public docs that it needs to be looked over carefully.

For the comments that are documenting a private interface, I'd suggest making them `///` comments: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use--for-public-quality-private-documentation . Then just write them as if they were public, with markdown and all.

Reply:
I agree with the argument that `//` serves as a reminder when we make it `///`. But with without backticks (or even square brackets), these reference can go stale anyways. I don't see a point of _not_ using them because `//` comments can get outdated. The `//` is sufficient as a reminder for re-checking them when publishing.